gobblin-runtime/src/test/java/org/apache/gobblin/runtime/api/FsSpecProducerTest.java

```diff
@@ -106,4 +131,19 @@ public void testDeleteSpec() throws URISyntaxException, ExecutionException, Inte
     Assert.assertEquals(jobSpecs.get(0).getLeft(), SpecExecutor.Verb.DELETE);
     Assert.assertEquals(jobSpecs.get(0).getRight().getUri().toString(), "testDeleteJob");
   }
+
+  class TestingFsSpecProducer extends FsSpecProducer {
```

Comment:
Can we simply write a dummy file (e.g. using File API) into the spec path without having to extend FsSpecProducer class? We can avoid having to make the annotateSpecFilename method package-private.

Reply:
Sure, I just realize `tmpDir` is available in the test. Addressed.